#158552546] ,#158527674  #158528050 Create test for API end points

LarryKarani opened the pull request:
**What does this PR do?**
This pull request create test for the API end points  _ GET/rides_ , _GET/rides/<rideId>_, _POST/rides_, _POST/rides/<rideId>/request_

**Description of Task to be completed?**
Test Api end points that do the following

- Get all ride offers

- Get a specific ride offer

- Create a ride offer

- Make a request to join a ride

**How the app should be tested**
- Clone the repo to your local machine
- Install the requirements specified in the requirements.txt file
- activate the virtual environment
- run _pytest_ in your cmd outside the tests directory

**Any background context you want to provide?**
All test fail the as no fuctionality has been implemented yet

**What are the relevant pivotal tracker stories?**
https://www.pivotaltracker.com/n/projects/2180294

**Screenshots of failing Test**
![test](https://user-images.githubusercontent.com/22986772/41817126-7966b48a-779d-11e8-9c45-eee7b7743bfd.PNG)
**

LarryKarani committed:
add test for create ride offer ap1, request ride api and the ride offer
 models with regards to TDD concepts finishes[#158552546]
 finishes[#158527674] finishes[#158528050]

JoshuaOndieki commented on FlaskApp/tests/test_rideoffers.py:
```diff
@@ -0,0 +1,39 @@
+
+    
+
+
```
Remove extra lines

JoshuaOndieki commented on FlaskApp/tests/test_routes.py:
```diff
@@ -0,0 +1,54 @@
+    def  test_make_request_to_join_valid_details(self):
+         self.response = self.dummy_client.post('/api/v1/rides/<rideId>/requests', data=json.dumps(self.request), content_type='application/json')
+         self.assertEqual(self.response.status_code, 201)
+         self.assertEqual(json.loads(self.response.data), Validator.validate(self.request))
```
**test_make_request_to_join_valid_details** doesn't sound like a good test name

LarryKarani committed:
Refactored the code to pass the writen test
 -add utils file that contains validatos which validate user input data
 -add API end points
 -add jwt authetification to secure endpoints
 finishes[#158527407] [#15828247] [#158527914] [#158528596]
